File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/ui/NetworkPagerAdapter.java
package com.didichuxing.doraemonkit.kit.network.ui;

import android.view.View;
import android.view.ViewGroup;

import androidx.viewpager.widget.PagerAdapter;

import com.didichuxing.doraemonkit.kit.network.bean.NetworkRecord;

import java.util.List;

public class NetworkPagerAdapter extends PagerAdapter {
    private List<NetworkDetailView> views;
    private NetworkRecord mRecord;

    public NetworkPagerAdapter(List<NetworkDetailView> views, NetworkRecord record) {
        this.views = views;
        mRecord = record;
    }

    @Override
    public int getCount() {
        return views.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        NetworkDetailView view = views.get(position);
        if (position == 0) {
            view.bindRequest(mRecord);
        } else {
            view.bindResponse(mRecord);
        }
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(views.get(position));
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/ui/NetworkPagerAdapter.kt
package com.didichuxing.doraemonkit.kit.network.ui

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import com.didichuxing.doraemonkit.kit.network.okhttp.bean.NetworkRecord

class NetworkPagerAdapter(private val views: List<NetworkDetailView>, private val mRecord: NetworkRecord) : PagerAdapter() {
    override fun getCount(): Int {
        return views.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = views[position]
        if (position == 0) {
            view.bindRequest(mRecord)
        } else {
            view.bindResponse(mRecord)
        }
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(views[position])
    }

}